### ISSUE TITLE
ultralytics randomly restarts training run while training

### ISSUE THREAD

**Issue by ThierryDeruyttere** [ultralytics/ultralytics#19448]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I'm having the weirdest problem. I'm training a yolov11 model on my data with the following params:

```
 epochs=50,
perspective=0.004,
shear=15,
imgsz=1024,
batch=0.6,
workers=2,
lr0=3e-4,
lrf=0.001,
cos_lr=True,
weight_decay=0.001,
model_type: str = "yolo11l-seg.pt",
optimizer: str = "auto",
fliplr: float = 0.5,
scale: float = 0,
imgsz: int = 1024, #: tuple = (768,1024),
profile: bool = False,
overlap_mask: bool = False,
degrees: float = 0,
```

Now at epoch 20ish or sometimes at other epochs, the training run randomly restarts without any error (see image). I'm wondering if someone ever had the same issue

<img width="1162" alt="Image" src="https://github.com/user-attachments/assets/a2f57fa3-1c72-467a-aaa5-f34f6a9579fb" />

### Environment

Linux machine on modal.com, 24GB gpu L4.

### Minimal Reproducible Example

No idea...

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ThierryDeruyttere, thank you for your interest in Ultralytics 🚀! We appreciate you reporting this issue and will assist you in resolving it.

To assist us in understanding and debugging this 🐛 Bug Report, could you please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) that demonstrates the issue? This will help us identify and address the problem more efficiently.

In the meantime, here are a few steps you could take to ensure the issue is not related to an outdated setup:
  
## Upgrade

Please upgrade to the latest `ultralytics` package, ensuring all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) are met in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

Your issue, occurring during training, could also be environment-related. Verify your environment is up to date and consider using one of our verified setups:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Debugging Suggestions

1. Verify the stability of your runtime environment on `modal.com`. Sometimes training instability can be due to external factors such as system interruptions or memory constraints.
2. Monitor GPU usage during training to identify potential resource issues.
3. Check and share any logs or warnings leading up to the restart—this might contain clues about the root cause.

## Community Support

You can also connect with the Ultralytics community for additional discussions:
- Join our [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time support.
- Share and learn on our [Discourse](https://community.ultralytics.com/) forum or [Subreddit](https://reddit.com/r/Ultralytics).

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing, confirming a stable codebase. CI tests verify YOLO functionality on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response, but rest assured, an Ultralytics engineer will review your report soon and provide further assistance. Thank you for contributing! 😊

**Comment by glenn-jocher** [comment]
@ThierryDeruyttere thanks for reporting this. To investigate further, please check your training logs for any OOM (Out of Memory) errors or interruptions. The `batch=0.6` setting attempts to use 60% of GPU memory, but memory spikes can still occur, especially with `imgsz=1024`. Try:  
1. Monitoring GPU memory usage in real-time with `nvidia-smi`  
2. Reducing `imgsz` or using `cache=True` to lower memory demands  
3. Testing with `workers=0` to eliminate data loading issues  

If the issue persists, please share full error logs and a reproducible example via a new GitHub issue. We'll be happy to investigate further.

**Comment by Y-T-G** [comment]
What's the training command? Are you running it through a script or CLI? What's the output of `yolo check`?
